legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -159,11 +138,43 @@ public Iterable<? extends Processor<?>> getExtraProcessors()
-                                    ._executable(context.pureModel.getPackageableElement(executable.executable.path, executable.executable.sourceInformation))) : null);
+                    {
+                        if (executableTitles.add(executable.title))
+                        {
+                            if (executable instanceof DataSpacePackageableElementExecutable)
+                            {
+                                return new Root_meta_pure_metamodel_dataSpace_DataSpacePackageableElementExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpacePackageableElementExecutable"))
+                                        ._title(executable.title)
+                                        ._description(executable.description)
+                                        ._executable(context.pureModel.getPackageableElement(((DataSpacePackageableElementExecutable) executable).executable.path, ((DataSpacePackageableElementExecutable) executable).executable.sourceInformation));
+                            }
+                            else if (executable instanceof DataSpaceTemplateExecutable)
+                            {
+
+
+                                    if (((DataSpaceTemplateExecutable) executable).executionContextKey != null && !dataSpace.executionContexts.stream().map(c -> c.name).collect(Collectors.toList()).contains(((DataSpaceTemplateExecutable) executable).executionContextKey))
+                                    {
+                                        throw new EngineException("Data space template executable's executionContextKey is not valid", dataSpace.sourceInformation, EngineErrorType.COMPILATION);
+                                    }
+                                    return new Root_meta_pure_metamodel_dataSpace_DataSpaceTemplateExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpaceTemplateExecutable"))
+                                            ._title(executable.title)
+                                            ._description(executable.description)
+                                            ._query(HelperValueSpecificationBuilder.buildLambda(((DataSpaceTemplateExecutable) executable).query, context))
+                                            ._executionContextKey(((DataSpaceTemplateExecutable) executable).executionContextKey);
+
+                            }
+                            else
+                            {
+                                throw new EngineException("Data space executables could only be template or executable", dataSpace.sourceInformation, EngineErrorType.COMPILATION);
+                            }
+                        }
+                        else
+                        {
+                            throw new EngineException("Data space executable title is not unique", dataSpace.sourceInformation, EngineErrorType.COMPILATION);
```

Comment:
can we inlcude the name of the title that was not unique.

Reply:
and same comment on source information. executable should have source info and should be filled out and provided with in error